test/test_basebackup_operation.py

```diff
@@ -125,5 +126,74 @@ def stream_handler(_stream):
         stream_handler=stream_handler,
         temp_dir=mysql_master.base_dir,
     )
-    with pytest.raises(Exception, match="^xtrabackup failed with code 13$"):
+    # we're opening a connection to check the version on mysql >= 8.0.29 below we're failing with the first message
+    with pytest.raises(Exception, match=r"(^xtrabackup failed with code 13$|^mysql_cursor\(\) missing 3 required keyword)"):
         op.create_backup()
+
+
+def test_backup_with_non_optimized_tables(mysql_master) -> None:
+    with myhoard_util.mysql_cursor(**mysql_master.connect_options) as cursor:
+        version = myhoard_util.get_mysql_version(cursor)
+        if version < "8.0.29":
+            raise SkipTest("DB version doesn't support OPTIMIZE TABLE")
+
+        def create_test_db(*, db_name: str, table_name: str, add_pk: bool) -> None:
+            cursor.execute(f"CREATE DATABASE {db_name}")
+
+            if add_pk:
+                id_column_type = "integer primary key"
+            else:
+                id_column_type = "integer"
+
+            cursor.execute(f"CREATE TABLE {db_name}.{table_name} (id {id_column_type})")
+            for value in range(15):
+                cursor.execute(f"INSERT INTO {db_name}.{table_name} (id) VALUES ({value})")
+            cursor.execute("COMMIT")
+            cursor.execute(f"ALTER TABLE {db_name}.{table_name} ADD COLUMN foobar VARCHAR(15)")
+            cursor.execute("COMMIT")
+
+        for db_index in range(15):
+            create_test_db(db_name=f"test{db_index}", table_name=f"foo{db_index}", add_pk=db_index % 2 == 0)
+
+        create_test_db(db_name="`söme/thing'; weird`", table_name="`table with space`", add_pk=True)
+
+        cursor.execute("SELECT NAME FROM INFORMATION_SCHEMA.INNODB_TABLES WHERE TOTAL_ROW_VERSIONS > 0")
+        tables_to_optimize = cursor.fetchall()
+        print(repr(tables_to_optimize))
+        assert len(tables_to_optimize) == 16
```

Comment:
This might be a hindrance once we build and use a version of MySQL that does not generate tables requiring optimization. The dance of updating the tests and the installed MySQL version at the same time could be painful.